ms2deepscore/data_generators.py

```diff
@@ -556,3 +518,41 @@ def __init__(self, spectrum_pair, tanimoto_score, dim, _data_augmentation, addit
             self.additional_inputs_right.append([float(self.spectrum_right.get(additional_input))])
 
         self.tanimoto_score = tanimoto_score
+
+
+def clean_reference_scores_df(reference_scores_df):
+    validate_labels(reference_scores_df)
+    reference_scores_df = exclude_nans_from_labels(reference_scores_df)
+    reference_scores_df = transform_to_inchikey14(reference_scores_df)
+    check_duplicated_indexes(reference_scores_df)
+    return reference_scores_df
+
+
+def validate_labels(reference_scores_df: pd.DataFrame):
+    if set(reference_scores_df.index) != set(reference_scores_df.columns):
+        raise ValueError("index and columns of reference_scores_df are not identical")
+
+
+def transform_to_inchikey14(reference_scores_df: pd.DataFrame):
+    """Transform index and column names from potential full InChIKeys to InChIKey14"""
+    reference_scores_df.index = [x[:14] for x in reference_scores_df.index]
+    reference_scores_df.columns = [x[:14] for x in reference_scores_df.columns]
+    return reference_scores_df
+
+
+def exclude_nans_from_labels(reference_scores_df: pd.DataFrame):
+    """Exclude nans in reference_scores_df, exclude columns and rows if there is any NaN
+    value"""
+    clean_df = reference_scores_df.dropna(axis='rows')  # drop rows with any NaN
+    clean_df = clean_df[clean_df.index]  # drop corresponding columns
+    n_dropped = len(reference_scores_df) - len(clean_df)
+    if n_dropped > 0:
+        print(f"{n_dropped} nans among {len(reference_scores_df)} labels will be excluded.")
+    return clean_df
+
+
+def check_duplicated_indexes(reference_scores_df):
```

Comment:
Are those functions used individually, outside `clean_reference_scores_df` or the generator classes?

If not, then we could add an underscore to those methods, just to indicate that those are merely helper functions

Reply:
No they are not. They were initially inside the classes as static functions. I just did not know that was a good habit for functions outside classes as well. So I will change that.